python_modules/dagster/dagster/_core/definitions/asset_check_execution.py

```diff
@@ -0,0 +1,113 @@
+import enum
+from typing import TYPE_CHECKING, NamedTuple, Optional
+
+import dagster._check as check
+
+from .asset_check_result import AssetCheckResult
+from .events import AssetKey
+
+if TYPE_CHECKING:
+    from dagster._core.event_api import EventLogRecord
+
+
+class AssetCheckExecutionStatus(enum.Enum):
+    PLANNED = "PLANNED"  # No result yet
+    RESULT_SUCCESS = "RESULT_SUCCESS"
+    RESULT_FAILURE = "RESULT_FAILURE"
+
+
+class AssetCheckExecution(
+    NamedTuple(
+        "_AssetCheckExecution",
+        [
+            ("asset_key", AssetKey),
+            ("check_name", str),
+            ("check_status", AssetCheckExecutionStatus),
+            ("run_id", str),
+            ("check_result", Optional[AssetCheckResult]),
+            ("target_materialization_record", Optional["EventLogRecord"]),
+            ("start_timestamp", Optional[float]),
+            ("end_timestamp", Optional[float]),
+        ],
+    )
+):
+    """Wrapper around AssetCheckResult that includes metadata about the execution of the check.
+
+    Normal use is to instantiate this class with AssetCheckExecution.planned(), then call
+    with_result() when the check is complete.
+    """
+
+    def __new__(
+        cls,
+        asset_key: AssetKey,
+        check_name: str,
+        check_status: AssetCheckExecutionStatus,
+        run_id: str,
+        check_result: Optional[AssetCheckResult] = None,
+        target_materialization_record: Optional["EventLogRecord"] = None,
+        start_timestamp: Optional[float] = None,
```

Comment:
could get this by querying the step, but I think it's helpful to denormalize so we don't have to do a join